### DISCUSSION TITLE
Data downloaded, but file is 0kb.

### DISCUSSION THREAD

**Discussion by bhups2k** [Leo4815162342/dukascopy-node#173] [Q&A]
I'm trying to download data but the file is 0KB in size and also has no extension.
Could someone please advise.

PS C:\temp> dir D:\download\

    Directory: D:\download

Mode                 LastWriteTime         Length Name
----                 -------------         ------ ----
-a----        26/04/2024     12:39          16534 BID_candles_min_1.bi5
-a----        26/04/2024     12:30              0 xauusd-m1-bid-2020-01-01-2024-04-26T11
-a----        26/04/2024     12:51              0 xauusd-m1-bid-2024-04-24-2024-04-26T11
-a----        26/04/2024     12:34              0 xauusd-m1-bid-2024-04-25-2024-04-26T11
-a----        26/04/2024     12:18              0 xauusd-tick-2020-01-01-2024-04-26T11

`> npx
> dukascopy-cli -i xauusd -from 2024-04-24 -to 2024-04-27 -t m1 -f csv -dir D:\download -d

  dukascopy-node:cli:version 1.38.0 +0ms
  dukascopy-node:cli:nodejs v20.12.2 +0ms
  dukascopy-node:cli:os Windows_NT, 10.0.19045 (win32) +0ms
  dukascopy-node:cli:config {
  dukascopy-node:cli:config   input: {
  dukascopy-node:cli:config     instrument: 'xauusd',
  dukascopy-node:cli:config     dates: { from: 2024-04-24T00:00:00.000Z, to: 2024-04-27T00:00:00.000Z },
  dukascopy-node:cli:config     timeframe: 'm1',
  dukascopy-node:cli:config     priceType: 'bid',
  dukascopy-node:cli:config     utcOffset: 0,
  dukascopy-node:cli:config     volumes: false,
  dukascopy-node:cli:config     volumeUnits: 'millions',
  dukascopy-node:cli:config     ignoreFlats: true,
  dukascopy-node:cli:config     dir: 'D:\\download',
  dukascopy-node:cli:config     silent: false,
  dukascopy-node:cli:config     format: 'csv',
  dukascopy-node:cli:config     batchSize: 10,
  dukascopy-node:cli:config     pauseBetweenBatchesMs: 1000,
  dukascopy-node:cli:config     useCache: false,
  dukascopy-node:cli:config     cacheFolderPath: './.dukascopy-cache',
  dukascopy-node:cli:config     retryCount: 0,
  dukascopy-node:cli:config     failAfterRetryCount: true,
  dukascopy-node:cli:config     retryOnEmpty: false,
  dukascopy-node:cli:config     pauseBetweenRetriesMs: 500,
  dukascopy-node:cli:config     debug: true,
  dukascopy-node:cli:config     inline: false,
  dukascopy-node:cli:config     fileName: '',
  dukascopy-node:cli:config     dateFormat: '',
  dukascopy-node:cli:config     timeZone: ''
  dukascopy-node:cli:config   },
  dukascopy-node:cli:config   isValid: true,
  dukascopy-node:cli:config   validationErrors: []
  dukascopy-node:cli:config } +0ms
  dukascopy-node:cli:urls Generated 13 urls +0ms
  dukascopy-node:cli:urls [
  dukascopy-node:cli:urls   'https://datafeed.dukascopy.com/datafeed/XAUUSD/2024/03/24/BID_candles_min_1.bi5',
  dukascopy-node:cli:urls   'https://datafeed.dukascopy.com/datafeed/XAUUSD/2024/03/25/BID_candles_min_1.bi5',
  dukascopy-node:cli:urls   'https://datafeed.dukascopy.com/datafeed/XAUUSD/2024/03/26/00h_ticks.bi5',
  dukascopy-node:cli:urls   'https://datafeed.dukascopy.com/datafeed/XAUUSD/2024/03/26/01h_ticks.bi5',
  dukascopy-node:cli:urls   'https://datafeed.dukascopy.com/datafeed/XAUUSD/2024/03/26/02h_ticks.bi5',
  dukascopy-node:cli:urls   'https://datafeed.dukascopy.com/datafeed/XAUUSD/2024/03/26/03h_ticks.bi5',
  dukascopy-node:cli:urls   'https://datafeed.dukascopy.com/datafeed/XAUUSD/2024/03/26/04h_ticks.bi5',
  dukascopy-node:cli:urls   'https://datafeed.dukascopy.com/datafeed/XAUUSD/2024/03/26/05h_ticks.bi5',
  dukascopy-node:cli:urls   'https://datafeed.dukascopy.com/datafeed/XAUUSD/2024/03/26/06h_ticks.bi5',
  dukascopy-node:cli:urls   'https://datafeed.dukascopy.com/datafeed/XAUUSD/2024/03/26/07h_ticks.bi5',
  dukascopy-node:cli:urls   'https://datafeed.dukascopy.com/datafeed/XAUUSD/2024/03/26/08h_ticks.bi5',
  dukascopy-node:cli:urls   'https://datafeed.dukascopy.com/datafeed/XAUUSD/2024/03/26/09h_ticks.bi5',
  dukascopy-node:cli:urls   'https://datafeed.dukascopy.com/datafeed/XAUUSD/2024/03/26/10h_ticks.bi5'
  dukascopy-node:cli:urls ] +0ms
  dukascopy-node:cli:fetcher https://datafeed.dukascopy.com/datafeed/XAUUSD/2024/03/25/BID_candles_min_1.bi5 | 15.9 KB | network +0ms
  dukascopy-node:cli:fetcher https://datafeed.dukascopy.com/datafeed/XAUUSD/2024/03/26/04h_ticks.bi5 | 14.16 KB | network +0ms
  dukascopy-node:cli:fetcher https://datafeed.dukascopy.com/datafeed/XAUUSD/2024/03/26/03h_ticks.bi5 | 19.42 KB | network +0ms
  dukascopy-node:cli:fetcher https://datafeed.dukascopy.com/datafeed/XAUUSD/2024/03/26/06h_ticks.bi5 | 32.44 KB | network +0ms
  dukascopy-node:cli:fetcher https://datafeed.dukascopy.com/datafeed/XAUUSD/2024/03/26/01h_ticks.bi5 | 21.21 KB | network +0ms
  dukascopy-node:cli:fetcher https://datafeed.dukascopy.com/datafeed/XAUUSD/2024/03/26/00h_ticks.bi5 | 7.18 KB | network +0ms
  dukascopy-node:cli:fetcher https://datafeed.dukascopy.com/datafeed/XAUUSD/2024/03/26/02h_ticks.bi5 | 20.8 KB | network +0ms
  dukascopy-node:cli:fetcher https://datafeed.dukascopy.com/datafeed/XAUUSD/2024/03/26/05h_ticks.bi5 | 16.34 KB | network +0ms
  dukascopy-node:cli:fetcher https://datafeed.dukascopy.com/datafeed/XAUUSD/2024/03/26/07h_ticks.bi5 | 44.77 KB | network +0ms
  dukascopy-node:cli:fetcher https://datafeed.dukascopy.com/datafeed/XAUUSD/2024/03/24/BID_candles_min_1.bi5 | 16.15 KB | network +0ms
  dukascopy-node:cli:fetcher https://datafeed.dukascopy.com/datafeed/XAUUSD/2024/03/26/08h_ticks.bi5 | 55.44 KB | network +0ms
  dukascopy-node:cli:fetcher https://datafeed.dukascopy.com/datafeed/XAUUSD/2024/03/26/09h_ticks.bi5 | 31.21 KB | network +0ms
  dukascopy-node:cli:fetcher https://datafeed.dukascopy.com/datafeed/XAUUSD/2024/03/26/10h_ticks.bi5 | 28.61 KB | network +0ms
----------------------------------------------------
√ File saved: D:\download\xauusd-m1-bid-2024-04-24-2024-04-26T11:50.csv (172.87 KB)

Download time: 3.8s`

**Comment by bhups2k** [comment]
Strange, but its working now.... nothing changed on my part.

**Reply by Leo4815162342** [reply]
@bhups2k perhaps, the underlying dukascopy datafeed server was either returning empty data

**Reply by 4x** [reply]
I suspect it's because the colon in the file name: did the 0kb file name get truncated i.e. "xauusd-m1-bid-2024-04-24-2024-04-26T11" with no extension? Use the `-fn` option and choose a filename without a colon, if your `-to` is `now` or any time that is not exactly a round hour.
